NuRadioReco/detector/detector_base.py

```diff
@@ -0,0 +1,995 @@
+
+    This class provides functions for all relevant detector properties.
+    """
+
+    def __init__(self, source='json', json_filename='ARIANNA/arianna_detector_db.json',
+                 dictionary=None, assume_inf=True, antenna_by_depth=True):
+        """
+        Initialize the stations detector properties.
+        By default, a new detector instance is only created of none exists yet, otherwise the existing instance
+        is returned. To force the creation of a new detector instance, pass the additional keyword parameter
+        `create_new=True` to this function. For more details, check the documentation for the
+        `Singleton metaclass <NuRadioReco.utilities.html#NuRadioReco.utilities.metaclasses.Singleton>`_.
+        
+        Parameters
+        ----------
+        source : str
+            'json', 'dictionary' or 'sql'
+            default value is 'json'
+            if dictionary is specified, the dictionary passed to __init__ is used
+            if 'sql' is specified, the file 'detector_sql_auth.json' file needs to be present in this folder that
+            specifies the sql server credentials (see 'detector_sql_auth.json.sample' for an example of the syntax)
+        json_filename : str
+            the path to the json detector description file (if first checks a path relative to this directory, then a
+            path relative to the current working directory of the user)
+            default value is 'ARIANNA/arianna_detector_db.json'
+        assume_inf : Bool
+            Default to True, if true forces antenna models to have infinite boundary conditions, otherwise the antenna madel will be determined by the station geometry.
+        antenna_by_depth: bool (default True)
+            if True the antenna model is determined automatically depending on the depth of the antenna. This is done by
+            appending e.g. '_InfFirn' to the antenna model name.
+            if False, the antenna model as specified in the database is used.
+        create_new: bool (default:False)
+            Can be used to force the creation of a new detector object. By default, the __init__ will anly create a new
+            object of none already exists.
+        """
+        self._serialization = SerializationMiddleware()
+        self._serialization.register_serializer(DateTimeSerializer(), 'TinyDate')
+        if source == 'sql':
+            self._db = buffer_db(in_memory=True)
+        elif source == 'dictionary':
+            self._db = TinyDB(storage=MemoryStorage)
+            self._db.truncate()
+            stations_table = self._db.table('stations', cache_size=1000)
+            for station in dictionary['stations'].values():
+                stations_table.insert(station)
+            channels_table = self._db.table('channels', cache_size=1000)
+            for channel in dictionary['channels'].values():
+                channels_table.insert(channel)
+        else:
```

Comment:
Another mistake I've definitely made before is to forget about the `source` argument and called `Detector(path_to_detector_json)` - this (unexpectly) loads the Arianna detector. This could be fixed by changing this to another `elif source == 'json':` + perhaps an error emitted if `source` is not one of `json, dictionary` or `sql`.

(Thanks a lot for these changes by the way Christoph! I'm not sure if github etiquette is to point out these suggestions or just make commits to your PR? I can also do the latter)

Reply:
This check is already done in the "new" Detector class.

Unfortunately the ARIANNA detector is set as the default argument for the JSON filename, that's why it is loaded. You can see that I wanted to remove this default, but that would require changing the order of parameters, since `source` has a default as well. And may break a lot of things.